src/components/common/form/PhoneNumberInput/PhoneNumberInput.native.js

```diff
@@ -51,62 +35,60 @@ export default ({
     return foundCountry
   }, [])
 
-  const onChangeHandler = useCallback(
-    (number: string) => {
-      let countryCode = phoneInputRef.current.getCountryCode()
-      const isUserTypingCountryNow = number.startsWith('+')
+  const onChange = useCallback((number: string) => {
+    let countryCode = phoneInputRef.current.getCountryCode()
+    const isUserTypingCountryNow = number.startsWith('+')
 
-      if (isUserTypingCountry !== isUserTypingCountryNow) {
-        setUserTypingCountry(isUserTypingCountryNow)
-      }
+    if (isUserTypingCountry !== isUserTypingCountryNow) {
+      setUserTypingCountry(isUserTypingCountryNow)
+    }
 
-      let completeNumber = number
+    let completeNumber = number
 
-      if (isUserTypingCountryNow) {
-        const isSameCountry = number.startsWith(`+${countryCode}`)
+    if (isUserTypingCountryNow) {
+      const isSameCountry = number.startsWith(`+${countryCode}`)
 
-        if (!isSameCountry) {
-          const countryISO = findMatchedCountry(number)
+      if (!isSameCountry) {
+        const countryISO = findMatchedCountry(number)
 
-          if (countryISO) {
-            phoneInputRef.current.selectCountry(countryISO)
-          }
+        if (countryISO) {
+          phoneInputRef.current.selectCountry(countryISO)
         }
-      } else {
-        completeNumber = `+${countryCode}${number}`
       }
+    } else {
+      completeNumber = `+${countryCode}${number}`
+    }
 
-      onChange(completeNumber)
-    },
-    [onChange],
-  )
+    props.onChange(completeNumber)
+  })
 
   let countryCode = ''
-  let phoneNumber = value
+  let phoneNumber = props.value
 
   if (phoneInputRef.current) {
     countryCode = `+${phoneInputRef.current.getCountryCode()}`
   }
 
   if (!isUserTypingCountry) {
-    phoneNumber = value.replace(countryCode, '')
+    phoneNumber = props.value.replace(countryCode, '')
   }
 
   const handleSelectCountry = useCallback(() => {
-    onChange('')
+    props.onChange('')
     setUserTypingCountry(false)
   })
 
+  const { onBlur, onFocus, placeholder, onSubmitEditing, enablesReturnKeyAutomatically } = props
   const textProps = { onBlur, onFocus, placeholder, onSubmitEditing, enablesReturnKeyAutomatically }
 
   return (
     <PhoneInput
       ref={phoneInputRef}
       value={phoneNumber}
-      onChangePhoneNumber={onChangeHandler}
+      onChangePhoneNumber={onChange}
       onSelectCountry={handleSelectCountry}
-      style={style}
-      textStyle={textStyle}
+      style={props.style}
```

Comment:
It has been resolved by you on this PR.
https://github.com/GoodDollar/GoodDAPP/commit/32305691a553cad13769a8b05bfd819aa6f20180